gap_nrf51.go

```diff
@@ -72,6 +72,12 @@ func (a *Advertisement) Stop() error {
 	return makeError(errCode)
 }
 
+// SetServiceData sets the service data for the advertisement.
+func (a *Advertisement) SetServiceData(sd []ServiceDataElement) error {
+	// TODO: implement
+	return nil
```

Comment:
As long as these are unimplemented, I'd say they need to return an error.

Reply:
Good point. Changed everywhere that it is not yet implemented.